src/jvmTest/kotlin/io/github/detekt/sarif4k/SarifMergingTest.kt

```diff
@@ -0,0 +1,31 @@
+package io.github.detekt.sarif4k
+
+import org.junit.jupiter.api.Test
+import java.io.File
+import kotlin.test.assertEquals
+
+class SarifMergingTest {
+    @Test
+    fun `merge output from sarifs of the same tool`() {
+        testMerge("/output.sarif.json", "/input_1.sarif.json", "/input_2.sarif.json")
+    }
+
+    @Test
+    fun `merge output from sarifs with different tools`() {
+        testMerge(
+            "/output_different_tool.sarif.json",
+            "/input_1.sarif.json",
+            "/input_2_different_tool.sarif.json"
+        )
+    }
+
+    private fun testMerge(expectedOutputFile: String, vararg inputFiles: String) {
+        val inputs = inputFiles.map {
+            SarifSerializer.fromJson(File(SarifSerializerTest::class.java.getResource(it)!!.path).readText())
+        }
+
+        val actual = SarifSerializer.toMinifiedJson(inputs.reduce(SarifSchema210::merge))
+        val expected = File(SarifSerializerTest::class.java.getResource(expectedOutputFile)!!.path).readText()
+        assertEquals(expected.replace("\\s".toRegex(), ""), actual)
```

Comment:
Consider deserializing the expected output and using `.equals` of SarifSchema210.
Removing all whitespace can clobber descriptions, paths, names, etc in test files, which will become a problem when we have more tests.

Reply:
Output is a bit messy because of all the nulls, but you are right, that this is more reliable. Will do.

![Uploading image.png…]()